sno/init.py

```diff
@@ -43,200 +57,368 @@ def import_gpkg(ctx, geopackage, table, list_tables):
 
     check_git_user(repo=None)
 
-    import_from = ["GPKG", geopackage, None]
-    if table and not list_tables:
-        import_from[2] = table
-
     if not list_tables and repo_path:
         repo_path.mkdir(exist_ok=True)
 
     ctx.invoke(
         init,
         directory=str(repo_path),
-        import_from=tuple(import_from),
+        import_from=f'GPKG:{geopackage}',
+        table=table,
         do_checkout=False,
     )
 
 
-class ImportPath(click.Path):
-    def __init__(self, prefixes=("GPKG",), suffix_required=False, **kwargs):
-        params = {
-            "exists": True,
-            "file_okay": True,
-            "dir_okay": False,
-            "writable": False,
-            "readable": True,
-            "resolve_path": False,
-            "allow_dash": False,
-            "path_type": None,
-        }
-        params.update(kwargs)
-
-        super().__init__(**params)
-
-        self.prefixes = prefixes
-        self.suffix_required = suffix_required
-
-    def convert(self, value, param, ctx):
-        if ":" not in value:
-            self.fail(f'expecting a prefix (eg. "GPKG:my.gpkg")')
-        prefix, value = value.split(":", 1)
+class OgrImporter:
+    """
+    Imports from an OGR source, currently from a whitelist of formats.
+    """
 
-        # need to deal with "GPKG:D:\foo\bar.gpkg:table"
-        search_from = 2 if is_windows else 0
+    @classmethod
+    def adapt_source_for_ogr(cls, source):
```

Comment:
suggest we add `OGR:{whatever}` which is passed directly without whitelisting?

Reply:
_could_ do... Is the goal to 'unofficially' support things other than what's been tested?

Reply:
Basically yes

Reply:
done in e2fe09acbfc727938bc12a087a6bb9eebfb9a5de